modal/_serialization.py

```diff
@@ -355,6 +369,10 @@ def serialize_data_format(obj: Any, data_format: int) -> bytes:
     elif data_format == api_pb2.DATA_FORMAT_GENERATOR_DONE:
         assert isinstance(obj, api_pb2.GeneratorDone)
         return obj.SerializeToString(deterministic=True)
+    elif data_format == api_pb2.DATA_FORMAT_CBOR:
+        if cbor2 is None:
+            raise InvalidError("CBOR support requires the 'cbor2' package to be installed.")
+        return cbor2.dumps(obj)
```

Comment:
Maybe we should catch `CBOREncodeTypeError`s here and give more context to the error message?

Reply:
Good idea, I'll add a test for that too